ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -476,26 +490,39 @@ private RowData(final long ts, final Struct key, final GenericRow value) {
 
     private final SqlType fieldType;
     private final ColumnName fieldName;
+    private final LogicalSchema schema;
     private final SqlValueCoercer defaultSqlValueCoercer = new DefaultSqlValueCoercer();
-
-    ExpressionResolver(final SqlType fieldType, final ColumnName fieldName) {
+    private final FunctionRegistry functionRegistry;
+    private final KsqlConfig config;
+
+    ExpressionResolver(
+        final SqlType fieldType,
+        final ColumnName fieldName,
+        final LogicalSchema schema,
+        final FunctionRegistry functionRegistry,
+        final KsqlConfig config
+    ) {
       this.fieldType = Objects.requireNonNull(fieldType, "fieldType");
       this.fieldName = Objects.requireNonNull(fieldName, "fieldName");
+      this.schema = Objects.requireNonNull(schema, "schema");
+      this.functionRegistry = Objects.requireNonNull(functionRegistry, "functionRegistry");
+      this.config = Objects.requireNonNull(config, "config");
     }
 
     @Override
-    protected String visitExpression(final Expression expression, final Void context) {
-      throw new KsqlException(
-          "Only Literals are supported for INSERT INTO. Got: "
-              + expression + " for field " + fieldName);
-    }
-
-    @Override
-    protected Object visitLiteral(final Literal node, final Void context) {
-      final Object value = node.getValue();
-      if (node instanceof NullLiteral || value == null) {
-        return null;
-      }
+    protected Object visitExpression(final Expression expression, final Void context) {
```

Comment:
I could do a benchmark, but I honestly think that would be overkill. Insert values was developed as a developer tool, not something to use in production environments and its performance reflects that (e.g. it creates a new kakfa producer for every message because that was easier to code up) so I think there are much bigger fish to fry if we wanted to make this "production ready"

Reply:
Ok, if it's not intended to be used to insert large amounts of data from scripts, then it doesn't seem so bad ;)